src/xtgeo/grid3d/grid_property.py

```diff
@@ -845,58 +963,81 @@ def ensure_correct_values(self, ncol, nrow, nlay, invalues):
     # Import and export
     # ==================================================================================
 
-    def from_file(self, pfile, fformat=None, **kwargs):  # _roffapiv for devel.
+    # TODO: Remove this ignore once import cycles cleaned up
+    @deprecation.deprecated(
+        deprecated_in="2.16",
+        removed_in="4.0",
+        current_version=xtgeo.version,  # type: ignore
+        details="Use xtgeo.gridproperty_from_file() instead",
+    )
```

Comment:
mypy takes issue with import `xtgeo.version` for now.

These deprecations are added far too late, but should be still far enough in advance that we can deprecate them whenever we hit version 4.